gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobufeditablelayer.cpp

```diff
@@ -0,0 +1,256 @@
+            new OGRFeature(poFlatGeobufTmpLayer->GetLayerDefn());
+        poNewFeature->SetFrom(poFeature.get(), aoMapSrcToTargetIdx.data(), true);
+        eErr = poFlatGeobufTmpLayer->CreateFeature(poNewFeature);
+        delete poNewFeature;
+    }
+    delete poFlatGeobufTmpLayer;
+
+    // Restore filters.
+    poEditableLayer->SetAttributeFilter(pszQueryStringBak);
+    CPLFree(pszQueryStringBak);
+    poEditableLayer->SetSpatialFilter(iFilterGeomIndexBak, poFilterGeomBak);
+    delete poFilterGeomBak;
+
+    if( eErr != OGRERR_NONE )
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "Error while creating %s",
+                 osTmpFilename.c_str());
+        VSIUnlink(osTmpFilename);
+        VSIUnlink(CPLResetExtension(osTmpFilename, "fgbt"));
+        return eErr;
+    }
+
+    delete m_poFlatGeobufLayer;
+
+    if( osFilename != osTmpFilename )
+    {
+        const CPLString osTmpOriFilename(osFilename + ".ogr_bak");
+        if (VSIRename(osFilename, osTmpOriFilename) != 0 ||
+            VSIRename(osTmpFilename, osFilename) != 0)
+        {
+            CPLError(CE_Failure, CPLE_AppDefined, "Cannot rename files");
+            *ppoDecoratedLayer = nullptr;
+            m_poFlatGeobufLayer = nullptr;
+            return OGRERR_FAILURE;
+        }
+        VSIUnlink(osTmpOriFilename);
+    }
+
+    VSILFILE *fp = VSIFOpenL(osFilename, "rb+");
+    if( fp == nullptr)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "Cannot reopen updated %s",
+                 osFilename.c_str());
+        *ppoDecoratedLayer = nullptr;
+        m_poFlatGeobufLayer = nullptr;
+        return OGRERR_FAILURE;
+    }
+
+    m_poFlatGeobufLayer = new OGRFlatGeobufLayer(osLayerName.c_str(), osFilename.c_str(), spatialRef, gType, createIndex, m_papszOpenOptions);
+    *ppoDecoratedLayer = m_poFlatGeobufLayer;
+
+    return OGRERR_NONE;
+}
+
+OGRFlatGeobufEditableLayer::OGRFlatGeobufEditableLayer(OGRFlatGeobufLayer *poFlatGeobufLayer,
+                                         char **papszOpenOptions) :
+    OGREditableLayer(poFlatGeobufLayer, true,
+                     new OGRFlatGeobufEditableLayerSynchronizer(
+                         poFlatGeobufLayer, papszOpenOptions),
+                     true)
+{
+    m_poFlatGeobufLayer = poFlatGeobufLayer;
+}
+
+OGRErr OGRFlatGeobufEditableLayer::CreateField( OGRFieldDefn *poNewField, int bApproxOK )
+{
+    if( m_poEditableFeatureDefn->GetFieldCount() >= 10000 )
+    {
+        CPLError(CE_Failure, CPLE_AppDefined,
+                 "Limiting to 10000 fields");
+        return OGRERR_FAILURE;
+    }
+
+    if( m_oSetFields.empty() )
+    {
+        for( int i = 0; i < m_poEditableFeatureDefn->GetFieldCount(); i++ )
+        {
+            m_oSetFields.insert(CPLString(
+                m_poEditableFeatureDefn->GetFieldDefn(i)->GetNameRef()).toupper());
+        }
+    }
+
+    const OGRFlatGeobufCreateFieldAction eAction = CREATE_FIELD_DO_NOTHING;
+    //OGRFlatGeobufLayer::PreCreateField(
```

Comment:
to remove all this CREATE_FIELD_ logic and the related enum. Must be CSV specific